### PR TITLE
RUN-1392: fix events recorder crashes

### PR DESCRIPTION
**Pull request by Domiii**
Fix events.

* -> 95 TestPassed vs. 6 TextFailed
  * https://github.com/replayio/build-test-dashboard/blob/master/06c174ff-3dc7-4362-aef4-eb4da45bd337
* https://linear.app/replay/issue/RUN-1392/fix-events-recorder-crashes

### CODE REVIEW

File: third_party/blink/renderer/bindings/core/v8/record_replay_events.cc [hunk 1]
```diff
@@ -85,10 +85,12 @@ UserEventProbe::UserEventProbe(const char* name,
     : name_(name ? String(name) : atomic_name),
       event_target_(event_target),
       is_callback_(is_callback) {
-  // Disabled by default, see https://linear.app/replay/issue/RUN-1251
   if (IsRecordingOrReplaying() &&
-      !FeatureEnabled("disable-collect-events") &&
-      !AreEventsDisallowed()) {
+      !AreEventsDisallowed() &&
+      // // Disabled by default (RUN-1251)
+      // !FeatureEnabled("disable-collect-events") &&
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_events.cc]: rm commented out code

**Review by bhackett1024 — CHANGES_REQUESTED** [review]
It looks like UserEventProbe::~UserEventProbe needs to be fixed too.

File: third_party/blink/renderer/bindings/core/v8/record_replay_events.cc (hunk 1 above)
Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_events.cc]: You read it too fast. It was already fixed 10 seconds later 👀

**Comment by Domiii** [comment]
> It looks like UserEventProbe::~UserEventProbe needs to be fixed too.

@bhackett1024, ouch - not sure how I overlooked that. On it.

**Review by kannanvijayan — APPROVED** [review]

File: third_party/blink/renderer/bindings/core/v8/record_replay_events.cc [hunk 2]
```diff
@@ -85,33 +92,22 @@ UserEventProbe::UserEventProbe(const char* name,
...
+    : UserEventProbe(name, atomic_name, event_target, !event_target) {}
 
+// NOTE: This is used for capturing script run events but it does not work.
 UserEventProbe::UserEventProbe()
-    // NOTE: This is used for capturing script run events but it does not work.
-    // TODO: uncomment - https://linear.app/replay/issue/RUN-1271
-    // UserEventProbe("scriptFirstStatement", AtomicString(), nullptr, false) 
-    {}
+// TODO: uncomment - https://linear.app/replay/issue/RUN-1271
+// UserEventProbe("scriptFirstStatement", AtomicString(), nullptr, false)
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_events.cc:106]: rm

**Review by bhackett1024 — APPROVED** [review]

File: third_party/blink/renderer/bindings/core/v8/record_replay_events.cc (hunk 2 above)
Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_events.cc:106]: This code should be in as-is. It cannot be different from this. I left a TODO as to what we have to fix to make it work.

Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_events.cc:106]: I'll just remove it, although i don't agree with the removal 🤔

**Comment by Domiii** [comment]
After my latest changes, I ran another test and got 14 `TestFailed` but 0 recorder crashes ([see here](https://ui.honeycomb.io/replay/datasets/backend/result/cDU8pLdtBTn/a/xsuXLJav6ee)). Merging.

**Pull request merged**